### PR TITLE
[Intel MKL] Factor out the macros for handling test registeration

### PR DESCRIPTION
**Pull request by CuiYifeng**

### CODE REVIEW

**Comment by gbaned** [comment]
@penpornk Can you please review this PR ? Thanks!

File: tensorflow/core/common_runtime/mkl_layout_pass_test.cc [hunk 1]
```diff
@@ -188,27 +189,6 @@ REGISTER_OP("BFloat16Output2")
...
-#define REGISTER_TEST_ALL_TYPES(TEST) \
-  REGISTER_TEST_FLOAT32(TEST);        \
-  REGISTER_TEST_BFLOAT16(TEST);
-#else
-#define REGISTER_TEST_ALL_TYPES(TEST) REGISTER_TEST_FLOAT32(TEST);
-#endif  // ENABLE_INTEL_MKL_BFLOAT16
-
-/////////////////////////////////////////////////////////////////////
-//  Unit tests related to node merge optimization
-/////////////////////////////////////////////////////////////////////
```
Comment (penpornk) [inline tensorflow/core/common_runtime/mkl_layout_pass_test.cc:210]: I think these are supposed to remain in this file. Would you mind moving it back?

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR and I'm sorry for the delay!

File: tensorflow/core/common_runtime/mkl_layout_pass_test.cc (hunk 1 above)
Comment (CuiYifeng) [reply tensorflow/core/common_runtime/mkl_layout_pass_test.cc]: Thanks for the reminder! I have moved it back.

**Review by penpornk — APPROVED** [review]
Thank you!

**Pull request merged**